### DISCUSSION TITLE
Problems importing SolidWorks or NX CAD files with NamedSelections

### DISCUSSION THREAD

**Discussion by aglowinski-danfoss** [ansys/pyansys-geometry#1474] [Q&A] [answered]
Dear PyAnsys Developers,
we want to use PyAnsys Geometry to translate and manipulate CAD assemblies for later use in PyPrime and Fluent.
Unfortunately we haven't managed to import the assembly including the Named Selection, yet.
We would appreciate some help here. Maybe this would also make a good example for the docs.
Best regards,
Alex

**Comment by jonahrb** [comment]
How are you creating the named selections? I'm honestly not sure that we support named selections directly from SolidWorks or NX without any use of Workbench. You could always just create the named selections in your PyAnsys Geometry script after you import (Example: https://geometry.docs.pyansys.com/version/stable/examples/01_getting_started/04_modeling.html#Create-a-named-selection).

**Comment by aglowinski-danfoss** [comment]
Then you might be right. I am using the named selection manager, that comes as a plugin for SolidWorks from Workbench. 
For our users it would not be a good solution, to create the Named Selections in PyAnsys, so I guess, we cannot use it for now.

**Reply by RobPasMue** [reply]
Hi @aglowinski-danfoss - quick question, and just to understand better the situation, why wouldn't it be possible to create the Named Selection using PyAnsys Geometry? We have an example in which we are doing precisely this: https://geometry.docs.pyansys.com/version/stable/examples/04_applied/02_naca_fluent.html#Create-named-selections

Is there any limitation you see? I'd just like to understand and see what we could do on our side to improve the user's experience with PyAnsys Geometry =)

Thanks in advance!

**Reply by aglowinski-danfoss** [reply]
Hi @RobPasMue ,
Let me explain you our use case. We have an automated PyFluent simulation script that can handle any assembly as long as it has the same named selections in it. That is the bottleneck here. For example we have one solidworks part, where we have 20 different named selections for volumes and surfaces, so I cannot not guarantee, that the same volume belongs to the same named selection.
A workaround might be to output a list of volumes and surface for every Named Selection, but I don't favor this solution.
Best regards,
Alex

**Reply by aglowinski-danfoss** [reply]
Another comment on that: As discussed before I am using my existing script and run it in SpaceClaim with:
modeler = launch_modeler_with_spaceclaim(product_version=241)
modeler.run_discovery_script_file(file_path='script.py')
This seems to throw me into a spaceclaim version, where I can only save my file with the scdocx extension, not the scdoc extension.
What am I missing here?

**Reply by RobPasMue** [reply, marked as answer]
That is expected - using PyAnsys Geometry you have several export methods which you can see in this link: https://geometry.docs.pyansys.com/version/stable/examples/03_modeling/export_design.html#Export-the-design

However, if you want to do it directly on the GUI, you will need to specify a dedicated environment variable on your system called ``SPACECLAIM_MODE`` and set it to 0 or 1. PyAnsys Geometry will set it to 2 by default if it is not defined by the user on env variables, which uses the Parasolid backend. For exporting to scdoc, you need the ACIS backend specified (0 or 1, I'm currently not sure). This is **only** available starting on 0.7.4 version.
